creed/rosetta.py

```diff
@@ -0,0 +1,290 @@
+                    io.imsave(save_path, comp_data[j, :, :, k], check_contrast=False)
+
+# TODO: Decide if this code is necessary or not
+# def compare_comped_images(raw_dir, comp_dir_list, output_dir):
+#     """Creates a tiled image containing the raw image, difference image, and output image
+#     from each compensated directory supplied
+#
+#     Args:
+#         raw_dir: directory containing raw images
+#         comp_dir_list: list of directories containing compensated images
+#         output_dir: directory where tifs will be saved"""
+#
+#     # load images
+#     raw_images = load_imgs_from_tree(raw_dir, dtype='float')
+#     comp_dict = {}
+#     for dir_name in comp_dir_list:
+#         comp_images = load_imgs_from_tree(dir_name, dtype='float')
+#         comp_dict[dir_name] = comp_images
+#
+#     img_size = raw_images.shape[1]
+#     fov_num = raw_images.shape[0]
+#     # compute difference between first compensation image and baseline
+#     diff_images = raw_images.values - comp_dict[comp_dir_list[0]].values
+#     diff_images[diff_images < 0] = 0
+#
+#     comp_image_num = len(comp_dir_list)
+#     # loop over each channel
+#     for j in range(raw_images.shape[3]):
+#         # create tiled array of corrections x fovs
+#         tiled_image = np.zeros((img_size * (comp_image_num + 2),
+#                                 img_size * fov_num))
+#         # loop over each fov, and place into columns of tiled array
+#         for i in range(raw_images.shape[0]):
+#             start = i * img_size
+#             end = (i + 1) * img_size
+#
+#             # first row is raw image
+#             tiled_image[:img_size, start:end] = raw_images.values[i, :, :, j]
+#
+#             # second row is difference image between raw and first compensated image
+#             tiled_image[img_size:(img_size * 2), start:end] = diff_images[i, :, :, j]
+#
+#             # subsequent rows are compensated images with different coefficients
+#             for idx, key in enumerate(comp_dir_list):
+#                 tiled_image[(img_size * (idx + 2)):(img_size * (idx + 3)), start:end] = \
+#                     comp_dict[key].values[i, :, :, j]
+#
+#         io.imsave(os.path.join(output_dir, raw_images.channels.values[j] + '_comparison.tiff'),
+#                   tiled_image)
+
+
+def create_tiled_comparison(input_dir_list, output_dir):
+    """Creates a tiled image comparing FOVs from all supplied runs for each channel.
+
+    Args:
+        input_dir_list: list of directories to compare
+        output_dir: directory where tifs will be saved"""
+
+    # load images
+    dir_dict = {}
+    dir_shapes = []
+    for dir_name in input_dir_list:
+        dir_images = load_imgs_from_tree(dir_name)
+        dir_shapes.append(dir_images.shape)
+        dir_dict[dir_name] = dir_images
+
+    if not np.all([shape == dir_shapes[0] for shape in dir_shapes]):
+        raise ValueError("All directories must contain the same number of images")
```

Comment:
I don't think we need `dir_dict` and `dir_shapes` for this and can simply validate using something along the lines of:

```
# get the number of images in the first FOV, would need to import list_files from io_utils
dir_len = len(list_files(input_dir_list[0], substr='.tiff'))

if not np.all([len(list_files(dir_len, substr='.tiff') == dir_len for dir_name in input_dir_list]):
    raise ValueError("All directories must contain the same number of images")
```

Reply:
If we didn't need `dir_dict` at all, I agree this would be a good fix. But if we're going to use it, I don't think this is any simpler than getting the shape as we loop through